src/algo/depth-to-rgb-calibration/optimizer.cpp

```diff
@@ -0,0 +1,1720 @@
+
+   /* double mat[3][4] = {
+        fx*(double)r[0] + ppx * (double)r[2], fx*(double)r[3] + ppx * (double)r[5], fx*(double)r[6] + ppx * (double)r[8], fx*(double)t[0] + ppx * (double)t[2],
+        fy*(double)r[1] + ppy * (double)r[2], fy*(double)r[4] + ppy * (double)r[5], fy*(double)r[7] + ppy * (double)r[8], fy*(double)t[1] + ppy * (double)t[2],
+        r[2], r[5], r[8], t[2] };
+*/
+    auto mat = p_mat.vals;
+    for( auto i = 0; i < z_data.vertices.size(); ++i )
+    {
+        double x = v[i].x;
+        double y = v[i].y;
+        double z = v[i].z;
+
+        double x1 = (double)mat[0] * (double)x + (double)mat[1] * (double)y + (double)mat[2] * (double)z + (double)mat[3];
+        double y1 = (double)mat[4] * (double)x + (double)mat[5] * (double)y + (double)mat[6] * (double)z + (double)mat[7];
+        double z1 = (double)mat[8] * (double)x + (double)mat[9] * (double)y + (double)mat[10] * (double)z + (double)mat[11];
+
+        auto x_in = x1 / z1;
+        auto y_in = y1 / z1;
+
+        auto x2 = ((x_in - ppx) / fx);
+        auto y2 = ((y_in - ppy) / fy);
+
+        f1[i].x = x2;
+        f1[i].y = y2;
+
+        auto r2 = (x2 * x2 + y2 * y2);
+
+        rc[i] = 1 + (double)yuy_intrin.coeffs[0] * r2 + (double)yuy_intrin.coeffs[1] * r2 * r2 + (double)yuy_intrin.coeffs[4] * r2 * r2 * r2;
+    }
+
+    return { f1,rc };
+}
+
+static p_matrix calc_gradients(
+    const z_frame_data& z_data,
+    const std::vector<double3>& new_vertices,
+    const yuy2_frame_data& yuy_data,
+    const std::vector<double2>& uv,
+    const calib & cal,
+    const p_matrix & p_mat,
+    data_collect * data = nullptr
+)
+{
+    p_matrix res;
+    auto interp_IDT_x = biliniar_interp( yuy_data.edges_IDTx, yuy_data.width, yuy_data.height, uv );      
+    auto interp_IDT_y = biliniar_interp( yuy_data.edges_IDTy, yuy_data.width, yuy_data.height, uv );
+
+    auto rc = calc_rc( z_data, new_vertices, yuy_data, cal, p_mat);
+
+    if (data)
+    {
+        data->iteration_data_p.d_vals_x = interp_IDT_x;
+        data->iteration_data_p.d_vals_y = interp_IDT_y;
+        data->iteration_data_p.xy = rc.first;
+        data->iteration_data_p.rc = rc.second;
+    }
+        
+    res = calc_p_gradients( z_data, new_vertices, yuy_data, interp_IDT_x, interp_IDT_y, cal, p_mat, rc.second, rc.first, data );
+    return res;
+}
+
+std::pair<double, p_matrix> calc_cost_and_grad(
+    const z_frame_data & z_data,
+    const std::vector<double3>& new_vertices,
+    const yuy2_frame_data & yuy_data,
+    const calib & cal,
+    const p_matrix & p_mat,
+    data_collect * data = nullptr
+)
+{
+    auto uvmap = get_texture_map(new_vertices, cal, p_mat);
+    if( data )
+        data->iteration_data_p.uvmap = uvmap;
+
+    auto cost = calc_cost(z_data, yuy_data, uvmap, data ? &data->iteration_data_p.d_vals : nullptr );
+    auto grad = calc_gradients(z_data, new_vertices, yuy_data, uvmap, cal, p_mat, data);
+    return { cost, grad };
+}
+
+params::params()
```

Comment:
Please specify the source reference for maintainers

Reply:
Source reference?

Reply:
The source reference (in short) for the hard-coded values